httpserver/server.go

```diff
@@ -139,6 +148,51 @@ func (srv *Server) GetInternalRouter() http.Handler {
 	return mux
 }
 
+// basicAuthMiddleware enforces HTTP Basic Auth on admin routes.
+// Username must match cfg.AdminBasicUser and password must match cfg.AdminPasswordBcrypt (bcrypt hash).
+func (srv *Server) basicAuthMiddleware() func(http.Handler) http.Handler {
+	requiredUser := srv.cfg.AdminBasicUser
+	bcryptHash := srv.cfg.AdminPasswordBcrypt
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if srv.cfg.AdminAuthDisabled {
+				// pass-through when disabled (for local development only)
+				next.ServeHTTP(w, r)
+				return
+			}
+			// If no hash configured, deny access (secure-by-default)
+			if bcryptHash == "" {
+				w.Header().Set("WWW-Authenticate", "Basic realm=admin")
+				http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+				return
+			}
+
+			u, p, ok := r.BasicAuth()
+			if !ok {
+				w.Header().Set("WWW-Authenticate", "Basic realm=admin")
+				http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+				return
+			}
+
+			if requiredUser != "" && u != requiredUser {
+				w.Header().Set("WWW-Authenticate", "Basic realm=admin")
+				http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+				return
+			}
+
+			// Compare password to bcrypt hash
+			if err := bcrypt.CompareHashAndPassword([]byte(bcryptHash), []byte(p)); err != nil {
+				w.Header().Set("WWW-Authenticate", "Basic realm=admin")
+				http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+				return
+			}
```

Comment:
Would be easier to follow flow this way:

```suggestion
			deny := func() {
				w.Header().Set("WWW-Authenticate", "Basic realm=admin")
				http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
			}
			
			// If no hash configured, deny access (secure-by-default)
			if bcryptHash == "" {
				deny()
				return
			}

			u, p, ok := r.BasicAuth()
			if !ok {
				deny()
				return
			}

			if requiredUser != "" && u != requiredUser {
				deny()
				return
			}

			// Compare password to bcrypt hash
			if err := bcrypt.CompareHashAndPassword([]byte(bcryptHash), []byte(p)); err != nil {
				deny()
				return
			}
```

Reply:
I'd also drop the condition for `requiredUser != ""`, one might run with empty user, and non-empty password. It'll work just fine

Reply:
`bcryptHash == ""` check could also be dropped. If the hash is empty, it will fail `bcrypt.CompareHashAndPassword` anyway.

Reply:
great suggestions

Reply:
I did leave `if bcryptHash == ""` in since it makes it extra visibile and explicit that it will fail.